### PR TITLE
Reduce garbage from allocations in DeprecationLogger

### PR DESCRIPTION
**Pull request by ebadyano**
1. Setting length for formatWarning String to avoid AbstractStringBuilder.ensureCapacityInternal calls
2. Adding extra check for parameter array length == 0 to avoid unnecessarily creating StringBuilder in LoggerMessageFormat.format

Helps to narrow the performance gap in throughout for geonames benchmark (#37411) by 3%. For more details:  https://github.com/elastic/elasticsearch/issues/37530#issuecomment-462758384

Relates to #37530
Relates to #37411
Relates to #35754

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java [hunk 1]
```diff
@@ -259,7 +259,12 @@ public Void run() {
      * @return a warning value formatted according to RFC 7234
      */
     public static String formatWarning(final String s) {
-        return WARNING_PREFIX + " " + "\"" + escapeAndEncode(s) + "\"";
+
```
Comment (danielmitterdorfer) [inline server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java]: Nit: remove empty line.

File: server/src/main/java/org/elasticsearch/common/logging/LoggerMessageFormat.java [hunk 2]
```diff
@@ -40,7 +40,7 @@ public static String format(final String prefix, final String messagePattern, fi
         if (messagePattern == null) {
             return null;
         }
-        if (argArray == null) {
+        if ((argArray == null) || (argArray.length == 0)) {
```
Comment (danielmitterdorfer) [inline server/src/main/java/org/elasticsearch/common/logging/LoggerMessageFormat.java]: Nit: remove redundant parentheses.

**Review by danielmitterdorfer — COMMENTED** [review]
I left two nits but the actual changes look fine to me. Can you please also:

* Update the PR title so it better reflects the purpose of the change, e.g. "Reduce garbage in deprecation logger" gives readers a much better understanding what this change is about than a more generic title.
* Update the PR description to indicate how much we gain from that?

**Comment by ebadyano** [comment]
@danielmitterdorfer Thank you for the review!

**Pull request merged**